src/main/java/com/passpnu/passwordmanager/entity/PasswordEntity.java

```diff
@@ -26,16 +27,21 @@ public class PasswordEntity {
     @GeneratedValue(strategy= GenerationType.AUTO)
     private Long id;
 
-    @NotBlank
     @ManyToOne
-    @JoinColumn(name="service_id", nullable = false)
+    @JoinColumn(name="service_id", nullable = false, updatable = false, insertable = false)
     private ServiceEntity service;
 
+    @Column(name="service_id")
+    private Long serviceId;
```

Comment:
I am not 100% sure whether you need both `service` and `serviceId` when you can easily get one from the other

Reply:
fixed